src/tt_compat.cpp

```diff
@@ -0,0 +1,36 @@
+#include <lua.hpp>
+#include <lauxlib.h>
+#include <string>
+#include "tt_compat.h"
+
+char* SER_MARKER_MAP = (char*)"map";
+char* SER_MARKER_SEQ = (char*)"seq";
+
+// Variable is needed to avoid collisions in LUA_REGISTRY.
+static const char TT_NULL_ADDR = 't';
+
+/* Mimics tarantool's cjson NULL, catches and sets it as global variable for future usage.
+ * Internally it loads json library, then extracts the "null" value from it and stores it in a registry.
+*/
+static void tt_lua_createnull(lua_State *L) {
+    luaL_register(L, "json", 0);
+    lua_pushlightuserdata(L, (void *)&TT_NULL_ADDR);
+    lua_getfield(L, -2, "null");
+    lua_settable(L, LUA_REGISTRYINDEX);
+}
+
+void tt_lua_pushnull(lua_State *L) {
```

Comment:
how about add `inline` here and move to .h?

Reply:
I would rather not doing it. It relies on the internal variable(some random address that is used to add extracted NULL to the global LUA_REGISTRY), so inlining that would mean exposing that internal variable as well.

Anyway, why do we even care about manual inlining? That's pretty small program, and we can even enable LTO to allow compiler making yet another round of optimization in hope that it would eliminate function call itself.

In other words, I would prefer to delegate such a manual labor to the machine. Let's just enable the LTO, and with the bench inspect the gain.

Reply:
ok, you can use a macros instead, or moe this code in luasimdjson.cpp

Reply:
LTO doesn't help us here surprisingly, so you are totally right, we must inline it ourselves. I've reworked the code.

Reply:
Added inlining to the serialize mt pushing as well.